http/1.7/modules/ROOT/pages/http-polling-source-ref.adoc

```diff
@@ -0,0 +1,300 @@
+        </otherwise>
+    </choice>
+</flow>
+
+<flow name="watermarkIntoQueryParamsPollingFlow">
+    <http:polling-source config-ref="watermarkRequestConfig" path="watermark-into-query"
+        splitExpression="#[payload.items]" watermarkExpression="#[item.wm]">
+        <scheduling-strategy>
+            <fixed-frequency frequency="1" timeUnit="SECONDS"/>
+        </scheduling-strategy>
+        <http:polling-request-query-params >
+            <http:polling-request-query-param key="watermark" value="#[watermark default 0]" />
+        </http:polling-request-query-params>
+    </http:polling-source>
+    <logger message="#[payload]"/>
+</flow>
+----
+
+== Configure Idempotency Expression
+
+To avoid the concurrent execution of two flows with the same payload identification, configure idempotency for the HTTP *Polling Source* event source. The configuration ensures that payloads with the same ID are processed not concurrently.
+
+To configure idempotency for the event source, add an ID expression to retrieve the ID from the item, similar to the watermarking expression configuration. Use `payload` and `item` placeholders for the idempotency expression `idExpression`.
+
+The following XML configuration example shows that the first three items execute the flow concurrently, and the fourth item starts only its process after the first item, both with the same ID, finishes processing.
+
+[source,xml,linenums]
+----
+<flow name="identityWithoutWatermarkListenerFlow">
+    <http:listener config-ref="watermarkListenerConfig" path="/identity-no-watermark"/>
+    <set-payload value="#[output application/json --- {'items': [{'name': 'Rodro', 'value': 5}, {'name': 'Eze', 'value': 8}, {'name': 'MG', 'value': 7}, {'name': 'Rodro', 'value': 14}]}]"/>
+</flow>
+
+<flow name="identityWithoutWatermarkPollingFlow">
+    <http:polling-source config-ref="watermarkRequestConfig" path="identity-no-watermark"
+        splitExpression="#[payload.items]" idExpression="#[item.name]">
+        <scheduling-strategy>
+            <fixed-frequency frequency="1" timeUnit="HOURS"/>
+        </scheduling-strategy>
+    </http:polling-source>
+    <logger message="#[payload]"/>
+</flow>
+----
+
+
+
+== Configure Response Validator
+
+To consider whether the response is valid, configure a response validator, as you do with the HTTP *Request* operation.
+
+For more information, refer to the xref:http-request-ref#HTTP-Response-Validation[Configure Response Validator] in the HTTP *Request* operation documentation.
+
+The following XML configuration example shows that because the server always returns a status code of 301, and the response validator expects the default status code from 200 to 299, the HTTP response always fails. This indicates that the flow does not execute, and the payload is only considered valid when the response validator defines 301 as the status code (the splitting, watermarking and idempotency configurations are applied then):
```

Comment:
```suggestion
The following XML configuration example shows a situation in which the server always returns a status code of 301, when the response validator expects a default status code between 200 and 299 and, as a result, the HTTP response always fails. This indicates that the flow does not execute, and the payload is considered valid only when the response validator defines 301 as the status code (the splitting, watermarking, and idempotency configurations are applied then):
```
Is the parenthetical information at the end of this sentence really necessary? It refers to something that is not part of this example's purpose.